### ISSUE TITLE
ch 10 - broken horizontal mobile layouts

### ISSUE THREAD

**Issue by satsie** [saving-satoshi/saving-satoshi#1183]
I don't know if we care, but mobile in landscape/horizontal in chapter 10 is still a bit buggy. On the right half of the screen, I can't scroll up to see the input/output part of the exercise. It's possible we'll close this as a "won't fix" but I wanted to file it first so it wouldn't get lost.

@GBKS do you have any input on how this should (or shouldn't) be prioritized?

Horizontal
![Screenshot from 2024-11-15 13-29-01](https://github.com/user-attachments/assets/2ad5f2b9-6772-469e-a690-87e36f64b839)

Vertical
![Screenshot from 2024-11-15 13-28-27](https://github.com/user-attachments/assets/259d53e8-32a6-4f58-a40f-454e4683cffd)

Original comment where I noticed this issue: https://github.com/saving-satoshi/saving-satoshi/pull/1173#issuecomment-2479685286

**Comment by GBKS** [comment]
Yes, we should absolutely fix that. It looks like the scroll bar is in the wrong spot. The input/output area is currently a height-limited box with scrolling enabled. It should just auto-expand to whatever height it needs to be, and the full page should be scrollable as needed.

Some people only have phones and no other devices. It would be a bit screwy if we make the experience for them good all the way and then have them stuck in chapter 10 due to a scroll issue. Mobile users are also people 😜

The text in the tabs should also not break into multiple lines.

**Comment by satsie** [comment]
Thanks for the input Christoph! Just to clarify, even if mobile works on vertical, we should fix horizontal right? I realize how silly that question sounds but I have to ask 🤣🤣

**Comment by GBKS** [comment]
Yes, we should fix that scrolling. What I mentioned above should also fix it for the landscape/horizontal layout you are referring to.

**Comment by GBKS** [comment]
I took another quick peek at this since it came up again on the call yesterday. Basically, scrolling is not set up properly on this page. There are even some [magic numbers in the code](https://github.com/saving-satoshi/saving-satoshi/blob/c64e3f41c95bc947cb80a85dd70048cf8347d0b6/ui/lesson/TransactionsChallenge/index.tsx#L284), which is not a good practice since they break across various responsive layouts (or when users adjust text size, which is not uncommon). This can all be automatically calculated by the browser with the right use of flexbox.

I marked things up in the image below. Right now, the red box is a scrollable container. That one should adjust to whatever height it needs to be. Instead, the green box should be a scrollable container. That matches how other challenges are set up, and also resolves the scroll issue on mobile landscape (and elsewhere).

![image](https://github.com/user-attachments/assets/617087a3-1b92-4300-9e1e-cfac44ae7035)

Peace & love.
